Tasks/Src/RTOS_BPS.c

```diff
@@ -58,9 +58,49 @@ void RTOS_BPS_MutexPost(BPS_OS_MUTEX* mutex, BPS_OS_OPT options) {
     OSMutexPost(mutex, options, &err);
     assertOSError(err);
 }
-void RTOS_BPS_TaskCreate(void){
 
+/**
+ * @brief   Creates a task that will be handled by the OS
+ * @param   *p_tcb - pointer to the tcb
+ * @param   *p_name - pointer to task name
+ * @param   p_task - pointer to the task
+ * @param   *p_args - pointer to task function arguments
+ * @param   *prio - task priority
+ * @param   *p_stk_base - the stack
+ * @param   stk_size - size of the stack
+ * @param   *p_err - return error code
+ * @return  nothing to see here
+ */
+void RTOS_BPS_TaskCreate(
+    OS_TCB              *p_tcb,
+    BPS_OS_TASK_NAME    *p_name,
+    BPS_OS_TASK_PTR      p_task,
+    void                *p_arg,
+    BPS_OS_TASK_PRIO     prio,
+    CPU_STK             *p_stk_base,
+    BPS_OS_TASK_STK_SIZE stk_size
+    )
+    {
+        BPS_OS_ERR err;
+        OS_TCB *tcbptr = calloc(1, sizeof(OS_TCB));
+        CPU_STK *stkptr = calloc(stk_size, sizeof(unsigned int)); // using size of unstacked int because cpu stack is type defined as an array of unsigned int
+        OSTaskCreate(tcbptr, p_name, p_task, p_arg, prio, stkptr, WATERMARK_STACK_LIMIT, stk_size, 0, 10,(void *)0, OS_OPT_TASK_STK_CHK | OS_OPT_TASK_SAVE_FP, &err);
```

Comment:
Why use WATERMARK_STACK_LIMIT? Shouldn't you use the function parameter stk_size? We want to use all of the task stack we can.